lib/testing/fakeVis.cpp

```diff
@@ -120,10 +121,35 @@ fakeVis::fakeVis(Config& config, const string& unique_name, bufferContainer& buf
                     break;
             }
             if (j == bins.size())
-                test_pattern_value[i] = default_val;
+                test_pattern_value[i] = default_val * std::conj(default_val);
             else
-                test_pattern_value[i] = bin_values.at(j);
+                test_pattern_value[i] = bin_values.at(j) * std::conj(bin_values.at(j));
         }
+        DEBUG("Using test pattern mode %s with default value %f+%fj and %d frequency values",
+              mode.c_str(), default_val.real(), default_val.imag(), bins.size());
+    } else if (mode == "test_pattern_inputs") {
+        std::vector<cfloat> input_values =
+            config.get<std::vector<cfloat>>(unique_name, "input_values");
+        if (input_values.size() != num_elements) {
+            throw std::invalid_argument("fakeVis: lengths of input values ("
+                                        + std::to_string(input_values.size())
+                                        + ") and number of elements ("
+                                        + std::to_string(num_elements) + ") have to be equal.");
+        }
+
+        size_t num_prods = num_elements * (num_elements + 1) / 2;
+        if (num_prods < 3)
```

Comment:
What case is this catching?   Wouldn't `num_elements == 1` just have one product, i.e. it's autocorrelation?